src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -6083,7 +6141,7 @@ private static bool DoDnsCallback(IAsyncResult result, MultipleAddressConnectAsy
             return DoMultipleAddressConnectCallback(PostOneBeginConnect(context), context);
         }
 
-        private class MultipleAddressConnectAsyncResult : ContextAwareResult
+        private sealed class MultipleAddressConnectAsyncResult : ContextAwareResult
```

Comment:
Whether or not it's out of date, it's about publically-exposed types, which this is not.  For any internal-only types, in general I believe we should have a seal-by-default stance, and only unseal when needed.  Sealing has potential performance benefits as well as potential usability benefits around understanding the extensibility potential of a type.